### ISSUE TITLE
[Errno 19] No such device

### ISSUE THREAD

**Issue by naranek** [s0lst1c3/sentrygun#1]
I tried the sentrygun on my Rasb Pi B, but it crashes with an error: error: [Errno 19] No such device

I also tried entering a bogus interface name, but in that case it crashes even earlier.

The same thing happens with the Master and the Update branches.

```
root@kali:~/tools/sentrygun# python sentrygun.py --karma --whitelist whitelist.txt -i wlan0


 _______  _______  _       _________ _______           _______           _
(  ____ \(  ____ \( (    /|\__   __/(  ____ )|\     /|(  ____ \|\     /|( (    /|
| (    \/| (    \/|  \  ( |   ) (   | (    )|( \   / )| (    \/| )   ( ||  \  ( |
| (_____ | (__    |   \ | |   | |   | (____)| \ (_) / | |      | |   | ||   \ | |
(_____  )|  __)   | (\ \) |   | |   |     __)  \   /  | | ____ | |   | || (\ \) |
      ) || (      | | \   |   | |   | (\ (      ) (   | | \_  )| |   | || | \   |
/\____) || (____/\| )  \  |   | |   | ) \ \__   | |   | (___) || (___) || )  \  |
\_______)(_______/|/    )_)   )_(   |/   \__/   \_/   (_______)(_______)|/    )_)


                            Written by Gabriel "solstice" Ryan

                            gabriel@solstice.me


[Karma Sentry] Sending Probe Request 0 for: 2Y5PBR6NN1A4V3NOM8PG9A94
[*] 802.11 Probe Request: SSID=[2Y5PBR6NN1A4V3NOM8PG9A94], count=10
Process Process-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "sentrygun.py", line 194, in detect_karma_attacks
    probe_responses = sniffer.send_probe_requests(interface=interface, ssid=next_essid)
  File "/root/tools/sentrygun/sniffer.py", line 101, in send_probe_requests
    ProbeReq(ssid=ssid, interface='wlp3s0')
  File "/root/tools/sentrygun/sniffer.py", line 85, in ProbeReq
    sendp(pkt,iface=interface,count=count,inter=0.1,verbose=0)
  File "/usr/lib/python2.7/dist-packages/scapy/sendrecv.py", line 259, in sendp
    __gen_send(conf.L2socket(iface=iface, *args, **kargs), x, inter=inter, loop=loop, count=count, verbose=verbose, realtime=realtime)
  File "/usr/lib/python2.7/dist-packages/scapy/arch/linux.py", line 417, in __init__
    self.ins.bind((iface, type))
  File "/usr/lib/python2.7/socket.py", line 228, in meth
    return getattr(self._sock,name)(*args)
error: [Errno 19] No such device

root@kali:~/tools/sentrygun# iwconfig
wlan0     IEEE 802.11bgn  Mode:Monitor  Frequency:2.467 GHz  Tx-Power=20 dBm   
          Retry short limit:7   RTS thr:off   Fragment thr:off
          Power Management:off        

```

**Comment by s0lst1c3** [comment]
Hey Jarmo, looking into this. Just to make sure we aren't missing anything,
do you mind running the command 'ifconfig' and then pasting the output
here? Thanks! :)

**Comment by naranek** [comment]
Here it is. I have two cards, but both error the same way.

```
wlan0: flags=867<UP,BROADCAST,NOTRAILERS,RUNNING,PROMISC,ALLMULTI>  mtu 1500
        unspec F4-F2-6D-35-9A-9C-30-30-00-00-00-00-00-00-00-00  txqueuelen 1000  (UNSPEC)
        RX packets 5384472  bytes 1623526387 (1.5 GiB)
        RX errors 0  dropped 3337460  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

wlan1: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        ether f4:f2:6d:73:9f:cc  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

```

**Comment by naranek** [comment]
I tested this on a fresh install of raspbian and the same NIC, and I get the same error. 

Also - you could add in the documentation that the program needs tcpdump to work. It's no biggie, the error message is clear, but still :)

**Comment by naranek** [comment]
Also tested with another type of NIC and still get the same error.

**Comment by s0lst1c3** [comment]
Fixed in commit a631f058b8b013fadc614759b2f8ae37cba84b02, thought it would autoclose.
